pandas/core/algorithms.py

```diff
@@ -2193,3 +2193,28 @@ def make_duplicates_of_left_unique_in_right(
     """
     left_duplicates = unique(left[duplicated(left)])
     return right[~(duplicated(right) & isin(right, left_duplicates))]
+
+
+def re_sort_union_after_inputs(union_values, lvals, rvals) -> np.ndarray:
+    """
+    Elements from union_values are resorted after the ranking of the first occurrence in
+    lvals and then rvals, if element is not in lvals. All occurrences of this element
+    are placed at the spot of the first occurrence of this element.
+
+    Parameters
+    ----------
+    union_values: np.array which is a sorted union of lvals and rvals
+    lvals: np.ndarray of the left values which is ordered in front.
+    rvals: np.ndarray of the right values ordered after lvals.
+
+    Returns
+    -------
+    np.ndarray containing the resorted values from union_values
+    """
+    indexer = []
+    counts = dict(zip(*np.unique(union_values, return_counts=True)))
```

Comment:
isn't this basically value_counts?

Reply:
After looking into value counts, yes. Changed it